validator_client/src/duties_service.rs

```diff
@@ -648,6 +668,25 @@ async fn poll_beacon_attesters_for_epoch<T: SlotClock + 'static, E: EthSpec>(
             .into_iter()
             .filter(|duty| local_pubkeys.contains(&duty.pubkey))
             .filter(|duty| {
+                if duties_service.per_validator_metrics() {
+                    let validator_index = duty.validator_index;
+                    let slot = duty.slot;
+                    if let Some(existing_slot) =
+                        get_int_gauge(&ATTESTATION_DUTY, &[&validator_index.to_string()])
+                    {
+                        let existing = Slot::new(existing_slot.get() as u64);
+                        if existing < current_slot || slot < existing {
```

Comment:
Is this logic to try and stop the next epoch duties from overriding the current epoch duties?

If so, I think we might want to compare the epochs rather than the slots. I think there's a scenario where we re-org to a later slot (in the same epoch) but we don't update the metrics.

Perhaps I'm missing something?

Reply:
> Is this logic to try and stop the next epoch duties from overriding the current epoch duties

Yep this is what I wanted. For some reason I thought that you can re-org only to an earlier slot. 
I think I have fixed the conditions now; although I'm trying to simplify it to handle all cases (the reorg and overriding current epoch duty scenarios).